fflib/src/classes/fflib_QueryFactory.cls

```diff
@@ -487,6 +494,23 @@ public class fflib_QueryFactory { //No explicit sharing declaration - inherit fr
 		);	
 		return this;
     }
+	/**
+	 * Remove existing ordering and set a field to be sorted on.  This may be a direct field or a field 
+	 * related through an object lookup or master-detail relationship.
+	 * Use the set to store unique field names, since we only want to sort
+	 * by the same field one time.  The sort expressions are stored in a list
+	 * so that they are applied to the SOQL in the same order that they
+	 * were added in. 
+	 * @param fieldName The string value of the field to be sorted on
+	 * @param SortOrder the direction to be sorted on (ASCENDING or DESCENDING)
+	 * @param nullsLast whether to sort null values last (NULLS LAST keyword included).
+	**/  
+    public fflib_QueryFactory setOrdering(String fieldName, SortOrder direction, Boolean nullsLast){
+		order = new List<Ordering> {
```

Comment:
**ACTION: Remove code duplication by chaining overloads**
Code duplication is evil. There should be a single "main" setOrdering(..) which does the actual work of replacing the previous orderings with a new Ordering. The other overloads of the method should simply call this main one, marshalling the arguments into the expected format and adding sensible defaults.
```java
public fflib_QueryFactory setOrdering(Ordering o){
    this.order = new List<Ordering>{ o };
    return this;
}
public fflib_QueryFactory setOrdering(String fieldName, SortOrder direction, Boolean nullsLast){
    return setOrdering(new Ordering(getFieldToken(fieldName), direction, nullsLast);
}
public fflib_QueryFactory setOrdering(SObjectField field, SortOrder direction, Boolean nullsLast){
    return setOrdering(new Ordering(new QueryField(field), direction, nullsLast);
}
```